### PR TITLE
ReadMe update for github actions linter

### PR DESCRIPTION
**Pull request by afroman32**
# Description

This PR updates the ReadMe for the github actions linter. I added to the documentation how to decode the error messages that the linter shows. This should make it easier for people to update their code to satisfy the linter.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [ ] Yes
- [ ] No
- [x] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by stimsonc — COMMENTED** [review]
Looks like this will make things easier to understand! I had some linting errors on my PR and had trouble understanding what to do.

**Comment by WhyseRabbit** [comment]
Documentation looks better, cleaning up and clarifying.

**Review by ppmorgoun — APPROVED** [review]
Nice work Makoa! Remember that the title of the PR should be readable in plain english, e.g.: "Updated Readme for X" so as long as you change that you've got an approval for me

**Comment by TashaSkyUp** [comment]
Looks good to me! But maybe while we have this on our minds, is it possible to include instructions on how to deal with the linter when there are many hundreds of - and + lines in your PR's log?

**Comment by afroman32** [comment]
@TashaSkyUp If there are many hundreds of - and + in the log then that means the code is very much out of line with python styling guides. The options here would be for the linting suggestions to be ignored(the next time the action is triggered it will not throw an error for the same files if they are not changed), take the time to change them and implement better coding sytling practices or I could update the action so that code is auto formatted on the way to be pushed.

**Review by bofori20 — APPROVED** [review]
Looks good to me.

**Comment by TashaSkyUp** [comment]
@afroman32

Pycharm does not have an issue with my code stylings. superlinter goes far beyond normal pep8 guidelines.

and

I have yet to be convinced that style guides are anything more than forcing people to code in a manner that the lowest common denominator can read, or maybe a popularity contest at best. I would say that if writing code is a skill that one can become progressively better at, then reading code well is a skill that people can get progressively better at.

Look at it this way maybe; if there is some style that is "perfect" what are the chances that someone has already found it?

if we actually want to find it, then we need to allow people to be free to use their own styling, because just having 10 small groups yelling for attention about how their style is better is not near as good at exploring the entire space of possible style solutions.

For myself, for open source projects, I will indeed ignore style guides unless I'm new to the project and the style guide is explicitly stated. If I'm employed however that is a completely different story, I'll follow what guidelines are given, and probably suggest some loose guidelines if none are given.

Thank you for your candor in this matter, it is important for us all to give and receive constructive criticism skillfully.

**Comment by afroman32** [comment]
@TashaSkyUp 
I agree with you that they aren't perfect, but they do add a bit of standardization and structure which I think is necessary when the turnover on a project is as high as this one and most of the students working on this project are novices. We don't know a bunch about leaving readable code and it's easy to get excited and leave out documentation or thoroughly check that what we did will be effective.

I do think that people should be able to experiment but, there does need to be structure. This is why I didn't set it to automatically change people's code. There is also the ability to kind of ignore the linter, because if you push again with the same code it will pass the test.

Thank you for giving me more insight into your issue, I will look into scaling back the aggressiveness of the linter.
